foxpuppet/ui/browser/window.py

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from .tabbar import TabBar
+from .navbar import Navbar
+from foxpuppet.ui.windows import Windows
+from foxpuppet.ui.base_window import BaseWindow
+
+
+class Browser_Window(object):
+
+    def __init__(self, foxpuppet, *args, **kwargs):
+        BaseWindow()
+        self.windows = Windows(foxpuppet)
+
+        self.selenium = foxpuppet.selenium
+        self._navbar = None
+        self._tabbar = None
+
+    @property
+    def navbar(self):
+        self.selenium.switch_to()
+
+        if not self._navbar:
+            navbar = self.selenium.find_element_by_id('nav-bar')
```

Comment:
For other page object models (like for addons) I have seen that element ids have been set as constants at the top. I feel that this is better and gives faster access in case of ui changes in Firefox.